src/main/scala/is/hail/expr/FunctionRegistry.scala

```diff
@@ -2575,37 +2575,68 @@ object FunctionRegistry {
         result: [0, 2, 4, 6]
     """
   )(arrayHr(TTHr), arrayHr(TTHr))
-  registerMethod("[*:]", (a: IndexedSeq[Any], i: Int) => a.slice(i, a.length),
+  registerMethod("[*:]", (a: IndexedSeq[Any], i: Int) => a.slice(if (i < 0) a.length + i else i, a.length),
     """
-    Returns a slice of the array from the i*th* element (0-indexed) to the end.
+    Returns a slice of the array from the i*th* element (0-indexed) to the
+    end. Negative indices are interpreted as offsets from the end of the string.
```

Comment:
This is the indexed seq case. "Negative indices are interpreted as offsets from the last element in the array."